packages/tools/piw-utils-internal/src/widgets/usePositionObserver.ts

```diff
@@ -1,37 +1,39 @@
-import { useCallback, useRef, useState } from "react";
+import { useCallback, useEffect, useRef, useState } from "react";
 
-export function usePositionObserver(target: HTMLElement | null): [ClientRect | undefined, () => () => void] {
+export function usePositionObserver(target: HTMLElement | null, active: boolean): ClientRect | undefined {
     const [position, setPosition] = useState<ClientRect>();
 
     const onAnimationFrameHandler = useCallback(() => {
-        if (!target) {
-            return;
-        }
-
-        const newPosition: ClientRect = target.getBoundingClientRect();
+        const newPosition: ClientRect | undefined = target?.getBoundingClientRect();
 
         if (shouldUpdatePosition(newPosition, position)) {
             setPosition(newPosition);
         }
     }, [position, target]);
 
-    return [position, useAnimationFrameHandler(onAnimationFrameHandler)];
+    useAnimationFrameEffect(active ? onAnimationFrameHandler : undefined);
+
+    return position;
 }
 
-function useAnimationFrameHandler(callback: () => void) {
+function useAnimationFrameEffect(callback?: () => void): void {
     const requestId = useRef<number | undefined>();
 
     const onAnimationFrame = useCallback(() => {
-        callback();
+        callback!();
```

Comment:
We have a check in the next line.. so in this case callback will never be undefined.

Reply:
This is the code as far as I can see.
```
    const onAnimationFrame = useCallback(() => {
        callback!();
        requestId.current = window.requestAnimationFrame(onAnimationFrame);
    }, [callback]);
```

But instead of the exclamation I feel like we can also just do
```
    const onAnimationFrame = useCallback(() => {
        if (callback) {
                callback();
        }
       requestId.current = window.requestAnimationFrame(onAnimationFrame);
    }, [callback]);
```

Reply:
The `!` was meant to emphasize that `onAnimationFrame` itself is never called if the callback is undefined, which is the intended behaviour. We _could_ replace it with `callback?.()` (no if please :D), but I think it's fine like this.